### PR TITLE
Convert diagram demos to functionality approach

### PR DESCRIPTION
**Pull request by dxvladislavvolkov**

### CODE REVIEW

File: JSDemos/Demos/Diagram/AdvancedDataBinding/React/App.js [hunk 1]
```diff
@@ -5,95 +5,93 @@ import Diagram, {
...
+  return 'none';
+}
 
-  linkFromLineEndExpr() {
-    return 'none';
-  }
+export default function App() {
+  React.useEffect(() => {
+    // Initialize the data sources
+    orgItemsDataSource.load();
```
Comment (ivanblinov2k17) [inline JSDemos/Demos/Diagram/AdvancedDataBinding/React/App.js]: do we really need to load ArrayStore?

File: JSDemos/Demos/Diagram/ImagesInShapes/React/App.js [hunk 2]
```diff
@@ -5,34 +5,30 @@ import Diagram, {
...
-      key: 'id',
-      data: service.getOrgItems(),
-    });
-    this.orgLinksDataSource = new ArrayStore({
-      key: 'id',
-      data: service.getOrgLinks(),
-    });
-  }
+export default function App() {
+  React.useEffect(() => {
```
Comment (ivanblinov2k17) [inline JSDemos/Demos/Diagram/ImagesInShapes/React/App.js]: empty useeffect

File: JSDemos/Demos/Diagram/ItemSelection/React/App.js [hunk 3]
```diff
@@ -5,64 +5,55 @@ import Diagram, {
...
+      selectedItems = filteredItems.join(', ');
     }
-    this.setState({
-      selectedItemNames,
-    });
-  }
+    setSelectedItemNames(selectedItems);
+  }, [setSelectedItemNames]);
+
+  React.useEffect(() => {
```
Comment (ivanblinov2k17) [inline JSDemos/Demos/Diagram/ItemSelection/React/App.js]: empty useeffect

File: JSDemos/Demos/Diagram/NodesAndEdgesArrays/React/App.js [hunk 4]
```diff
@@ -5,33 +5,29 @@ import Diagram, {
...
-      data: service.getFlowNodes(),
-    });
-    this.flowEdgesDataSource = new ArrayStore({
-      key: 'id',
-      data: service.getFlowEdges(),
-    });
-  }
+export default function App() {
+  React.useEffect(() => {
+    // Any necessary side effects can be added here
```
Comment (ivanblinov2k17) [inline JSDemos/Demos/Diagram/NodesAndEdgesArrays/React/App.js]: same

File: JSDemos/Demos/Diagram/NodesArrayPlainStructure/React/App.js [hunk 5]
```diff
@@ -5,28 +5,24 @@ import Diagram, {
...
+});
 
-    this.dataSource = new ArrayStore({
-      key: 'ID',
-      data: service.getEmployees(),
-    });
-  }
+export default function App() {
+  React.useEffect(() => {
+    // Any side effects can be added here
```
Comment (ivanblinov2k17) [inline JSDemos/Demos/Diagram/NodesArrayPlainStructure/React/App.js]: same

File: JSDemos/Demos/Diagram/SimpleView/React/App.js [hunk 6]
```diff
@@ -5,27 +5,23 @@ import Diagram, {
 import ArrayStore from 'devextreme/data/array_store';
 import service from './data.js';
 
-class App extends React.Component {
-  constructor(props) {
-    super(props);
+export default function App() {
+  const [dataSource, setDataSource] = React.useState([]);
```
Comment (alexanderPolosatov) [inline JSDemos/Demos/Diagram/SimpleView/React/App.js]: dataSource can be moved from state

**Review by ivanblinov2k17 — APPROVED** [review]

**Pull request merged**